themes-default/slim/views/vue-components/name-pattern.mako

```diff
@@ -0,0 +1,491 @@
+                    type: this.type,
+                    multiEpStyle: this.selectedMultiEpStyle,
+                    custom: this.isCustom,
+                    enabled: this.isEnabled,
+                    animeNamingType: this.animeType
+                });
+            },
+            async checkNaming(pattern, selectedMultiEpStyle, animeType) {
+                const vm = this;
+                const params = {
+                    pattern: pattern,
+                    anime_type: animeType
+                }
+
+                if (selectedMultiEpStyle) {
+                    params.multi = selectedMultiEpStyle;
+                }
+
+                try {
+                    const result = await apiRoute.get('config/postProcessing/isNamingValid', {params: params});
+                    if (result.data === 'invalid') {
+                        $(vm.$el).find('#naming_pattern').qtip('option', {
+                            'content.text': 'This pattern is invalid.',
+                            'style.classes': 'qtip-rounded qtip-shadow qtip-red'
+                        });
+                        $(vm.$el).find('#naming_pattern').qtip('toggle', true);
+                        $(vm.$el).find('#naming_pattern').css('background-color', '#FFDDDD');
+                    } else if (result.data === 'seasonfolders') {
+                        $(vm.$el).find('#naming_pattern').qtip('option', {
+                            'content.text': 'This pattern would be invalid without the folders, using it will force "Flatten" off for all shows.',
+                            'style.classes': 'qtip-rounded qtip-shadow qtip-red'
+                        });
+                        $(vm.$el).find('#naming_pattern').qtip('toggle', true);
+                        $(vm.$el).find('#naming_pattern').css('background-color', '#FFFFDD');
+                    } else {
+                        $(vm.$el).find('#naming_pattern').qtip('option', {
+                            'content.text': 'This pattern is valid.',
+                            'style.classes': 'qtip-rounded qtip-shadow qtip-green'
+                        });
+                        $(vm.$el).find('#naming_pattern').qtip('toggle', false);
+                        $(vm.$el).find('#naming_pattern').css('background-color', '#FFFFFF');
+                    }
+
+                } catch (e) {
+                    console.warning(e);
+                }
+
+            }
+        },
+        computed: {
+            isCustom() {
+                return !this.presets.includes(this.pattern) || this.pattern === 'Custom...';
+            },
+            selectedNamingPattern: {
+                get: function() {
+                    return (this.isCustom) ? 'Custom...' : this.pattern;
+                },
+                set: function(value) {
+                    this.pattern = value;
+                }
+            }
+        },
+        mounted() {
+            this.pattern = this.namingPattern;
```

Comment:
Why aren't all of these set in data? The mounted shouldn't be setting default values.

Reply:
I'm setting the values for the local data using the properties. That's what you should do in mounted right? data() i am using to set the defaults.

Reply:
Why not set the default on the prop and then just pass `pattern` as a prop instead of `namingPattern` ?

Reply:
I can't directly use the properties, as then I would change them by reference. And Vue.js doesn't like that